docs/cli-architecture.md

```diff
@@ -0,0 +1,9 @@
+# LambdaBuffers CLI architecture
+
+LambdaBuffers consists of three runtime command line interface components:
+
+- [lambda-buffers-frontend-cli](lambda-buffers-frontend-cli)
```

Comment:
Do we intend to flesh this out for the deadline? Links are dead ATM (obviously)

On a less urgent note: Do we really need three separate cli tools here? We don't expect users to have to manually pipe output between the components, so presumably each of these cli components will be subsumed by some user-facing monolith, right?

Reply:
I'll describe the granular cli architecture here and leave the monolith-cli for later when we flesh out how exactly it should behave.